sdk/go/common/providers/reference.go

```diff
@@ -20,7 +20,6 @@ import (
 	"strings"
```

Comment:
I had to move this to `sdk/go/common`, as we are now using it from the new `VerifyIntegrity` function that works on `DeploymentV3`. Not sure if there's a better alternative. Another option would be to move `VerifyIntegrity` into `pkg`, but that doesn't match well with how these `apitype`s currently work (and it might even be useful to be able to run `VerifyIntegrity` on the service potentially?  

Happy to explore alternatives to this big refactor though.  Also happy to split it out into a separate PR if that's easier to review.

Reply:
I think that this is fine as far as a refactor. The WASM work also needs something like this in order to shrink the dependency closure.

> and it might even be useful to be able to run `VerifyIntegrity` on the service potentially

Agreed. Seems like it's generally good to be able to run these checks earlier / without too much deserialization.

Reply:
This was a breaking change.

https://github.com/search?q=org%3Apulumi%20IsProviderType&type=code

https://github.com/search?q=org%3Apulumi+IsDefaultProvider&type=code